aiida/orm/nodes/data/folder.py

```diff
@@ -19,3 +19,22 @@
 
 class FolderData(Data):
     """`Data` sub class to represent a folder on a file system."""
+
+    def __init__(self, **kwargs):
+        """Construct a new `FolderData` to which any files and folders can be added.
+
+        To add files to a new node use the various repository methods:
+
+            folder = FolderData()
+            folder.put_object_from_tree('/absolute/path/to/directory')
+            folder.put_object_from_filepath('/absolute/path/to/file.txt')
+            folder.put_object_from_filelike(filelike_object)
+
+        Alternatively, in order to simply wrap a directory, the `path` keyword can be used in the constructor:
```

Comment:
I would put this first, and then mention the other methods.
```suggestion
        Use the `tree` keyword in order to simply wrap a directory:
```